compiler/damlc/daml-ide-core/src/Development/IDE/Core/Rules/Daml.hs

```diff
@@ -904,19 +905,22 @@ toDiagnostic ::
     -> Range
     -> Either SS.Error SS.ScenarioResult
     -> Maybe FileDiagnostic
-toDiagnostic file world range (Left err) =
-    Just $
-    (file, ShowDiag, ) $
-    Diagnostic
+toDiagnostic file world range = \case
+    Left err -> Just $ mkDiagnostic DsError $ formatScenarioError world err
+    Right SS.ScenarioResult{..}
+        | V.null scenarioResultWarnings -> Nothing
+        | otherwise -> Just $ mkDiagnostic DsWarning $
+            LF.prettyWarningMessages scenarioResultWarnings
```

Comment:
is it worth making this a separate diagnostic for each warning? Slightly orthogonal it could be very useful to embed to use the commit location (the callsite of `submit`) instead of the location of the scenario for the warnings which makes it much easier to pin down.

Reply:
I like the idea of using the submit location. Is that known to the engine, or would it be something we have to figure out how to propagate?

(Merging for now, since it should be a separate PR in any case.)

Reply:
As for separate warnings, I think it makes sense to have separate them if we separate the location. I don't think there's much value in having separate warnings otherwise.

Reply:
It is known to the engine and we already display it in Daml Studio. You can scrape it from https://github.com/digital-asset/daml/blob/cf2f79db4eaaaaba73bd3f1ccd503d9afcb0f07a/daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Speedy.scala#L121

Reply:
Thanks!